src/CMakeLists.txt

```diff
@@ -1,15 +1,20 @@
 set(headers dlfcn.h)
 set(sources dlfcn.c)
 
-if (BUILD_SHARED_LIBS)
-  add_definitions(-DSHARED)
-endif (BUILD_SHARED_LIBS)
 
 add_library(dl ${sources})
 
 # Correctly export the location of installed includes in the target
 target_include_directories(dl INTERFACE $<INSTALL_INTERFACE:include>)
 
+# export the functions
+target_compile_definitions(dl PRIVATE DLFCN_WIN32_EXPORTS)
```

Comment:
This seems to be incorrect. Functions cannot be exported when building dlfcn as static library.